packages/components/src/internal/components/domainproperties/assay/AssayDesignerPanels.tsx

```diff
@@ -128,6 +132,34 @@ class AssayDesignerPanelsImpl extends React.PureComponent<Props & InjectedBaseDo
         return !appIsValidMsg ? undefined : appIsValidMsg(protocolModel);
     }
 
+    getTextChoiceUpdatesValidMsg(): string {
+        const { protocolModel } = this.state;
+
+        if (!protocolModel.editableRuns) {
+            const domain = protocolModel.getDomainByNameSuffix('Run');
+            const hasFieldValueUpdates =
+                domain.fields.find(field => {
+                    return field.textChoiceValidator?.extraProperties?.valueUpdates !== undefined;
+                }) !== undefined;
+            if (hasFieldValueUpdates) {
+                return 'Text choice value updates are not allowed when assay does not allow "Editable Runs".';
+            }
```

Comment:
This bit of logic could be factored out since it's virtually the same for the "run" and "data" cases.

Reply:
done